svg-slideshow.py

```diff
@@ -1,64 +1,77 @@
+    parser.add_argument('--compress', action='store_true', help='enable lossless compression. Will overwrite original file.')
+    return parser
+
+
+def add_image(img):
+    with open(img, "rb") as image_file:
+        buf = image_file.read()
+        fmt = imghdr.what(None, buf)
+        encoded = base64.b64encode(buf)
+        image = tree.SubElement(svg, 'image', opacity='1.0', width=str(w), height=str(h))
+        image.set(u'xlink:href', u"data:image/%s;base64,%s" % (fmt, encoded.decode("utf-8")))
+        return image
+
+
+if __name__ == '__main__':
+    parser = init_parser()
+    parser.print_help()
+    args = parser.parse_args()
+    print("starting......")
+    frames = len(args.files)
+    print("Trying to load %d images" % frames)
+
+    w = args.width
+    h = args.height
+    fade = args.fade
+    delay = args.delay
+    dur = (fade + delay) * frames
+
+    print("Slideshow will be %d ms" % dur)
+
+    with open(args.outfile, 'w') as f:
+        f.write('<?xml version="1.0" encoding="utf-8"?>\n')
+        svg = tree.Element('svg', width=str(w), height=str(h), version='1.1', xmlns='http://www.w3.org/2000/svg',
+                           viewBox='0 0 %d %d' % (w, h))
+        svg.set(u'xmlns:xlink', u'http://www.w3.org/1999/xlink')
+
+        add_image(args.files[0])
+        # append first image to back of list for a smooth transition
+        args.files.append(args.files.pop(0))
+        t = 0
+        for img in args.files:
+            if args.compress:
+                img_c = Image.open(img)
+                img_c.save(img, format=img_c.format, optimize=True)
+
+            print('Adding "%s"' % img)
+            t += delay
+            image = add_image(img)
+            animate = tree.SubElement(image, 'animate', attributeName='opacity', attributeType='XML',
+                                      repeatCount='indefinite', begin='0', dur='%dms' % dur,
+                                      keyTimes= "0.0;%f;%f;1.0" % (t/dur, (t + fade)/dur),
+                                      values="0.0;0.0;1.0;1.0", calcMode='linear')
+            t += fade
+
+        f.write(tree.tostring(svg, encoding="unicode"))
+
+
+#    add  Fallback if animate is not supported
 #    <image opacity="1.0" width="640" height="480" xlink:href="2017-11-28-235151_648x506_scrot.png">
-#    </image>
-
-  print('Adding "%s"' % p)
-  #encoded = bytes()
-
-  startImage(p)
-  f.write('  <animate attributeName="opacity" attributeType="XML"')
-  f.write('                 repeatCount="indefinite"')
-  f.write('                 begin="0"')
-  f.write('                 dur="%dms"' % dur)
-  t += delay
-  f.write('                 keyTimes="0;%f;%f;1.0"' % (t / dur, (t + fade) / dur))
-  t += fade
-  f.write('                 values="0.0;0.0;1.0;1.0"')
-  f.write('                 calcMode="linear" />\n')
-  f.write('</image>\n')
-
-  print("Processed %d / %d" % (t, dur))
-
-f.write('</svg>')
+#    </image>
```

Comment:
nit: Please re-add the newline? Some people like to `cat` their files / some editors expect it.